upgrade: Handle multi-node upgrade status

skazi0 opened the pull request:
During disruptive upgrade multiple nodes can be upgraded in prallel.
The status structure was adapted to this on the backend side.
This change updates the UI to match the backend.

**EDIT**
Final screenshots attached below in a separate comment.

1 node:
![zrzut ekranu z 2017-03-09 11-53-47](https://cloud.githubusercontent.com/assets/2458112/23747571/7776c24c-04c0-11e7-8de4-ca6bd8fd926b.png)
2-4 nodes:
![zrzut ekranu z 2017-03-09 11-53-22](https://cloud.githubusercontent.com/assets/2458112/23747569/7772ef96-04c0-11e7-86dd-54faefb45487.png)
&gt;4 nodes:
![zrzut ekranu z 2017-03-09 11-52-34](https://cloud.githubusercontent.com/assets/2458112/23747568/774fd056-04c0-11e7-91ca-e5894767ed6b.png)

skazi0 committed:
upgrade: Handle multi-node upgrade status

During disruptive upgrade multiple nodes can be upgraded in prallel.
The status structure was adapted to this on the backend side.
This change updates the UI to match the backend.

skazi0 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(8) S56P5 Implement disruptive upgrade (non-HA, one compute node, etc.)](https://trello.com/c/2B0jUWrs/334-8-s56p5-implement-disruptive-upgrade-non-ha-one-compute-node-etc)

vuntz commented:
I'm fine with it, but have two comments/questions:

 - can we write: "controller.1234.suse.com (role: controller)" instead of "controller.1234.suse.com / role: controller"
 - is there a way to get info on the "2 more" somehow? From the cli, maybe? I can imagine it'd be important in case things are hanging.

skazi0 commented:
@vuntz 

- as we always group nodes by role, we can even remove the role from those lines and add it as a separate line like in single node version.
- the information is visible in the server response and in status file, I don't think we trim the output in cli so it should be visible there too.

jsuchome commented:
Yes, CLI output shows always everything

skazi0 committed:
upgrade: Display node role only once and reorder

Role was removed from current node lines in multi-node display.
Instead, just like with single node the role is displayed as separate line.
The lines were reordered to better reflect importance.

skazi0 commented:
Final screenshots:
![zrzut ekranu z 2017-03-09 14-07-28](https://cloud.githubusercontent.com/assets/2458112/23751822/492a84ce-04d3-11e7-984d-a470b849900c.png)
![zrzut ekranu z 2017-03-09 14-08-09](https://cloud.githubusercontent.com/assets/2458112/23751823/492aec20-04d3-11e7-82a0-2e951e686d77.png)
